File: src/main/java/cn/mobiledaily/web/common/SpringContext.java
package cn.mobiledaily.web.common;

import cn.mobiledaily.service.AttendeeService;
import cn.mobiledaily.service.ExhibitionService;
import cn.mobiledaily.service.FileService;
import cn.mobiledaily.web.assembler.CheckInAssembler;

import javax.faces.context.FacesContext;

/**
 * Get Spring context bean via FacesContext
 */
public class SpringContext {
    private static <T> T getBean(String name, Class<? extends T> type) {
        FacesContext context = FacesContext.getCurrentInstance();
        return context.getApplication().evaluateExpressionGet(context, "#{" + name + "}", type);
    }

    public static ExhibitionService getExhibitionService() {
        return getBean("exhibitionService", ExhibitionService.class);
    }

    public static FileService getFileService() {
        return getBean("fileService", FileService.class);
    }

    public static AttendeeService getAttendeeService() {
        return getBean("attendeeService", AttendeeService.class);
    }

    public static CheckInAssembler getCheckInAssembler() {
        return getBean("checkInAssembler", CheckInAssembler.class);
    }
}

File: src/main/scala/cn/mobiledaily/web/common/SpringContext.scala
package cn.mobiledaily.web.common

import javax.faces.context.FacesContext
import cn.mobiledaily.service.{AttendeeService, FileService, ExhibitionService}
import cn.mobiledaily.web.assembler.CheckInAssembler

object SpringContext {
    private def get[T](name: String, clazz: Class[T]) = {
        val facesContext = FacesContext.getCurrentInstance
        facesContext.getApplication.evaluateExpressionGet(facesContext, "#{" + name + "}", clazz)
    }

    def exhibitionService = get("exhibitionService", classOf[ExhibitionService])

    def fileService = get("fileService", classOf[FileService])

    def attendeeService = get("attendeeService", classOf[AttendeeService])

    def checkInAssembler = get("checkInAssembler", classOf[CheckInAssembler])
}
